redisgraph/execution_plan.py

```diff
@@ -0,0 +1,75 @@
+class Operation:
+    def __init__(self, name, args=None):
+        self.name = name
+        self.args = args
+        self.children = []
+
+    def append_child(self, child):
+        if not isinstance(child, Operation) or self is child:
+            raise Exception("child must be Operation")
+
+        self.children.append(child)
+        return self
+
+    def __eq__(self, o: object) -> bool:
+        if not isinstance(o, Operation):
+            return False
+
+        if self.name != o.name or self.args != o.args or len(self.children) != len(o.children):
+            return False
+
+        for i in range(len(self.children)):
+            if not self.children[i] == o.children[i]:
+                return False
+
+        return True
+
+    def __str__(self) -> str:
+        args_str = "" if self.args is None else f" | {self.args}"
+        if len(self.children) == 0:
+            return f"{self.name}{args_str}"
+        else:
+            children = "\n".join(["    " + line for op in self.children for line in str(op).splitlines()])
+            return f"{self.name}{args_str}\n{children}"
```

Comment:
Operation to string should print only the `self` operation, traversal over the children should be performed by the `ExecutionPlan` object.

Reply:
done